src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/matcher/Matcher.kt

```diff
@@ -0,0 +1,14 @@
+package uk.gov.justice.digital.hmpps.personrecord.service.matcher
+
+import uk.gov.justice.digital.hmpps.personrecord.model.Person
+
+abstract class Matcher<T>(val items: List<T>?, protected val person: Person) {
```

Comment:
This could be an interface I think - it doesn't have any state

Reply:
I do really like this change by the way

Reply:
We could use an interface but I prefer this approach as this is simpler.